### PR TITLE
Fix Issue 23968 - Deprecation not emitted with alias to template function in UFCS

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

**Comment by RazvanN7** [comment]
While investigating this issue I found out that the way the compiler deals with deprecations on aliases is kind of weird. My expectation was that anytime an alias is substituted with the aliasee (via the `toAlias` call) the alias declaration would be checked for deprecations. However, I found out that the compiler is littered with something along the lines of `if (s.isAliasDeclaration()) checkDeprecation() ; s = s.toAlias()`. I haven't looked at all uses, but it seems to me that most deprecation checks could be pulled in the toAlias method of the AliasDeclaration.

Either way, this should be accepted as is and subsequent PRs should tackle this issue.

File: compiler/src/dmd/expressionsem.d [hunk 1]
```diff
@@ -489,6 +489,15 @@ private Expression searchUFCS(Scope* sc, UnaExp ue, Identifier ident)
 
     if (auto dti = ue.isDotTemplateInstanceExp())
     {
+        // https://issues.dlang.org/show_bug.cgi?id=23968
+        // Typically, deprecated alias declarations are catched
```
Comment (CyberShadow) [inline compiler/src/dmd/expressionsem.d:493]:
```suggestion
        // Typically, deprecated alias declarations are caught
```

**Review by CyberShadow — COMMENTED** [review]
Thank you!

**Comment by RazvanN7** [comment]
cc @dkorpel

**Review by dkorpel — APPROVED** [review]

**Pull request merged**